cmd/http/tracer.go

```diff
@@ -24,6 +32,39 @@ func tracerProvider(url string) (*trace.TracerProvider, error) {
 			semconv.ServiceNameKey.String(config.TraceName),
 			attribute.String("environment", config.TraceEnv),
 		)),
+		trace.WithSampler(sampler),
 	)
 	return tp, nil
 }
+
+type ProbabilisticSampler struct {
+	max      int
+	boundary int // values equal or greater than boundary would be dropped; 1/00 prob can be expressed as max=100, boundary=1
+}
+
+func (s ProbabilisticSampler) ShouldSample(p trace.SamplingParameters) trace.SamplingResult {
+	r := rand.Intn(s.max)
+	if r >= s.boundary {
+		return trace.SamplingResult{
+			Decision:   trace.Drop,
+			Tracestate: oteltrace.SpanContextFromContext(p.ParentContext).TraceState(),
+		}
+	}
+
+	return trace.SamplingResult{
+		Decision:   trace.RecordAndSample,
+		Tracestate: oteltrace.SpanContextFromContext(p.ParentContext).TraceState(),
+	}
+}
+
+func (s ProbabilisticSampler) Description() string {
+	return "probabilistic sampler"
+}
+
+func NewProbabilisticSampler(probability float64) *ProbabilisticSampler {
+	const max = 1000.0
```

Comment:
suggestion: upper case for const variable

golangprograms.com/go-language/constants.html#:~:text=Naming%20Conventions%20for%20Golang%20Constants,usually%20written%20in%20uppercase%20letters.

Reply:
changed as suggested

Reply:
replaced the arbitrary 1000 with max.MaxInt32